src/lib.rs

```diff
@@ -19,47 +19,68 @@ fn take_guess() -> char {
         .expect("More than one character was guessed.");
 }
 
-fn run_game_loop(config: &mut Config, input: char) -> GameState {
-    if !config.secret_word.contains(input) {
-        return GameState::LOST;
+struct PlayerState {
+    wrong_guesses: u32,
+}
+
+fn run_game_loop(config: &mut Config, guess: char, player_state: &mut PlayerState) -> GameState {
+    if !config.secret_word.contains(guess) {
+        player_state.wrong_guesses += 1;
+
+        if player_state.wrong_guesses > config.lives {
+            return GameState::LOST;
+        } else {
+            println!(
+                "WRONG... lives remaining {}",
+                config.lives - player_state.wrong_guesses
+            );
+            return GameState::PLAYING;
+        }
     }
 
+    update_correctly_guessed_letters(config, guess);
+
+    println!("CORRECT... status: {}", config.correctly_guessed_letters);
+
+    if config.correctly_guessed_letters == config.secret_word {
+        return GameState::WON;
+    }
+
+    // return GameState::PLAYING;
+    return GameState::PLAYING;
+}
+
+fn update_correctly_guessed_letters(config: &mut Config, guess: char) {
     // check for hits of guess in solution
     let hit_idxs: Vec<usize> = config
```

Comment:
In this function we are mutating the `config` object. What do you think about making the `correctly_guessed_letters` part of the `PlayerState` (possibly in a follow-up PR)?

Reply:
already changed that way in later PR!